### PR TITLE
feat: implementing moler for inference.

### PR DESCRIPTION
**Pull request by drugilsberg**
Implementing MoLeR inference pipeline: https://github.com/microsoft/molecule-generation

### CODE REVIEW

File: src/gt4sd/algorithms/generation/moler/implementation.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self._seed = seed
+
+    def generate(self) -> List[str]:
+        """Sample molecules using MoLeR.
+
+        Returns:
+            sampled molecule (SMILES).
+        """
+        # process scaffolds
+        valid_scaffolds = [
```
Comment (jannisborn) [inline src/gt4sd/algorithms/generation/moler/implementation.py:77]: This only makes sure the scaffolds are molecules but not that they are scaffolds.
We could use:

```py
if (
    Chem.MolToSmiles(Chem.MolFromSmiles(scaffold), canonical=True) ==
    Chem.Scaffolds.MurckoScaffold.MurckoScaffoldSmiles(scaffold)
)
```

I think the murcko scaffold canonicalizes under the hood so we need to account for that too,

**Review by jannisborn — COMMENTED** [review]
Interesting paper, so I like the effort to integrate it. It looks great, I just left one comment about validating the scaffolds.

The tests are failing, it seems that the weights are not yet on the COS, at least the tests seem to not find them.

Also, if I see correctly, the `list_versions` method is not yet implemented even though the tests are trying to use it.

File: src/gt4sd/algorithms/generation/moler/implementation.py (hunk 1 above)
Comment (drugilsberg) [reply src/gt4sd/algorithms/generation/moler/implementation.py:77]: I think they have an imprecise naming in their function, they indeed accept any mol object as scaffolds following their call stack

**Comment by drugilsberg** [comment]
> 

exactly I had to jump in some calls, but I will put them on COS soon.

File: src/gt4sd/algorithms/generation/moler/implementation.py (hunk 1 above)
Comment (jannisborn) [reply src/gt4sd/algorithms/generation/moler/implementation.py:77]: Looking at their paper, this is a bit surprising but we should probably stick to the original implementation.

**Review by jannisborn — APPROVED** [review]

**Pull request merged**